drivers/display/display_st7735r.c

```diff
@@ -545,6 +546,7 @@ static const struct display_driver_api st7735r_api = {
 		.frmctr3 = DT_INST_PROP(inst, frmctr3),				\
 		.gamctrp1 = DT_INST_PROP(inst, gamctrp1),			\
 		.gamctrn1 = DT_INST_PROP(inst, gamctrn1),			\
+		.swap_color_order = DT_INST_PROP(inst, swap_color_order),	\
```

Comment:
What is the purpose of swap-color-order property since it does not change anything?

Reply:
New property does not change the behavior of the controller, therefore it makes no sense anyway. Why can you not set correct value to `madctl` property (RGB-BGR ORDER)?

Reply:
@jfischer-no , @ck-telecom 

> New property does not change the behavior of the controller, therefore it makes no sense anyway. Why can you not set correct value to madctl property (RGB-BGR ORDER)?

Shortly say, to work around bugs in the LCD module.

The built-in LCD module of Longan Nano that I'm trying to support with #44796 has a problem that is RGB/BGR setting is not match to actual displaying.

I created a pattern for color confirmation with such a code and confirmed the actual display of Longan Nano. 

```
        for(int xx=0; xx<16; xx++) {
                uint16_t* buf16 = (uint16_t*)buf;
                for(int j=0; j<capabilities.x_resolution/16*capabilities.y_resolution; j++) {
                        uint16_t col = (1U << xx);
                        *buf16++ = (col&0xFF) << 8 | (col&0xFF00)>>8;
                }
                display_write(display_dev, (capabilities.x_resolution/16)*(15-xx), 0, &buf_desc, buf);
        }
```
We can see the bit and the corresponding color in this pattern.

MADCTL is 112 (order is RGB)

![IMG-3592](https://user-images.githubusercontent.com/458281/163408800-01af29bd-ea99-4855-8d2e-78ac8ca77fbc.jpg)

the pixel format is 🟦🟦🟦🟦🟦🟩🟩🟩🟩🟩🟩🟥🟥🟥🟥🟥

(white square indicate (0,0))

MADCTL is 120 (order is BGR)

![IMG-3591](https://user-images.githubusercontent.com/458281/163408835-31933848-6c34-42c0-8aef-83abd30722fa.jpg)

the pixel format is 🟥🟥🟥🟥🟥🟩🟩🟩🟩🟩🟩🟦🟦🟦🟦🟦

The pixel format settings of the controller and the actual display are mismatched.
(I'm guessing there's a problem connecting the st7735 controller to the LCD panel.)

We can draw correctly by fitting the pixel format to match the actual display, but there is one problem here.

The st7735r driver gets the pixel format of the display from the MADCTL settings. On the Longan Nano LCD, the RGB/BGR settings of the controller and displaying of that are different. So the correct pixel format cannot obtain from the capabilities structure.

By swapping the RGB/BGR order of MADCTL and the RGB/BGR order that can obtain by API, that can be satisfied the API specifications and the actual displaying.

The above is the purpose of introducing this property.